### PR TITLE
Set up new repository

### PR DESCRIPTION
**Pull request by mkaput**
> [!IMPORTANT]
> Review commit-by-commit. I want to rebase these commits over `main` instead of squashing

### CODE REVIEW

**Review by Draggu — APPROVED** [review]

**Review by piotmag769 — APPROVED** [review]

**Review by piotmag769 — APPROVED** [review]

**Review by Draggu — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -2,170 +2,77 @@ name: CI
...
-          - test --profile=ci -p cairo-lang-syntax-codegen
-          - test --profile=ci -p cairo-lang-test-plugin
-          - test --profile=ci -p cairo-lang-test-runner
-          - test --profile=ci -p cairo-lang-test-utils
-          - test --profile=ci -p cairo-lang-utils --features=serde,parity-scale-codec,schemars,testing,env_logger
-          - test --profile=ci -p cairo-lang-utils --no-default-features --features=serde,parity-scale-codec
-          - test --profile=ci -p tests
+        platform:
+          - name: linux x86-64
+            os: ubuntu-latest
```
Comment (Arcticae) [inline .github/workflows/ci.yml:19]: Does it need to be in the matrix though?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -2,170 +2,77 @@ name: CI
...
       - uses: actions/checkout@v4
       - uses: dtolnay/rust-toolchain@master
         with:
           toolchain: nightly-2024-11-23
+          components: rustfmt
+      - uses: dtolnay/rust-toolchain@stable
       - uses: Swatinem/rust-cache@v2
-      - run: >
-          scripts/docs.sh
+      - run: cargo +nightly-2024-11-23 fmt --check
```
Comment (Arcticae) [inline .github/workflows/ci.yml:65]: why do we need the nightly here

File: scripts/clippy.sh
Comment (Arcticae) [inline scripts/clippy.sh:1]: Don't we want to have this script?

File: src/project/unmanaged_core_crate.rs [hunk 3]
```diff
@@ -33,13 +33,6 @@ pub fn try_to_init_unmanaged_core(
 pub fn find_unmanaged_core(config: &Config, scarb: &ScarbToolchain) -> Option<PathBuf> {
     find_core_at_config_path(config)
         .or_else(|| find_scarb_managed_core(scarb))
-        .or_else(|| {
-            if cfg!(feature = "testing") {
-                cairo_lang_filesystem::detect::detect_corelib()
-            } else {
-                None
-            }
```
Comment (Arcticae) [inline src/project/unmanaged_core_crate.rs:41]: Does it need to be changed here? The dep doesn't contain the corelib and that's the problem here?

**Review by Arcticae — CHANGES_REQUESTED** [review]

File: Cargo.toml [hunk 4]
```diff
@@ -13,20 +13,20 @@ testing = []
...
-cairo-lang-lowering = { git = "https://github.com/starkware-libs/cairo", rev = "3c5a2f414204388da4db6f9530cd22e6d1ba5184" }
-cairo-lang-parser = { git = "https://github.com/starkware-libs/cairo", rev = "3c5a2f414204388da4db6f9530cd22e6d1ba5184" }
-cairo-lang-project = { git = "https://github.com/starkware-libs/cairo", rev = "3c5a2f414204388da4db6f9530cd22e6d1ba5184" }
-cairo-lang-semantic = { git = "https://github.com/starkware-libs/cairo", rev = "3c5a2f414204388da4db6f9530cd22e6d1ba5184" }
-cairo-lang-starknet = { git = "https://github.com/starkware-libs/cairo", rev = "3c5a2f414204388da4db6f9530cd22e6d1ba5184" }
-cairo-lang-syntax = { git = "https://github.com/starkware-libs/cairo", rev = "3c5a2f414204388da4db6f9530cd22e6d1ba5184" }
-cairo-lang-test-plugin = { git = "https://github.com/starkware-libs/cairo", rev = "3c5a2f414204388da4db6f9530cd22e6d1ba5184" }
-cairo-lang-utils = { git = "https://github.com/starkware-libs/cairo", rev = "3c5a2f414204388da4db6f9530cd22e6d1ba5184" }
+cairo-lang-compiler = { git = "https://github.com/starkware-libs/cairo", rev = "6512b4bf99d49afd6d50129ee5d1f8f4f6fbc308" }
```
[the remark this answers is not in the document]
Comment (mkaput) [reply Cargo.toml:16]: we're not going to always have a tag while we're in middle of release cycle. see how it's looking in Scarb

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -25,42 +25,42 @@ jobs:
...
-          - test --profile=ci-dev -p cairo-lang-starknet
-          - test --profile=ci-dev -p cairo-lang-syntax
-          - test --profile=ci-dev -p cairo-lang-syntax-codegen
-          - test --profile=ci-dev -p cairo-lang-test-plugin
-          - test --profile=ci-dev -p cairo-lang-test-runner
-          - test --profile=ci-dev -p cairo-lang-test-utils
-          - test --profile=ci-dev -p cairo-lang-utils --features=serde,parity-scale-codec,schemars,testing,env_logger
-          - test --profile=ci-dev -p cairo-lang-utils --no-default-features --features=serde,parity-scale-codec
-          - test --profile=ci-dev -p tests
+          - test --profile=ci -p cairo-lang-casm --features=serde,parity-scale-codec,schemars
```
[the remark this answers is not in the document]
Comment (mkaput) [reply .github/workflows/ci.yml]: see following commits, I'm dropping these lines

File: Cargo.toml [hunk 6]
```diff
@@ -27,64 +23,3 @@ edition = "2021"
 repository = "https://github.com/starkware-libs/cairo/"
 license = "Apache-2.0"
 license-file = "LICENSE"
-
-[workspace.dependencies]
```
[the remark this answers is not in the document]
Comment (mkaput) [reply Cargo.toml:83]: I do not plan to have any except xtasks which have an isolated dependency set

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,27 +1,38 @@
...
 
-If your commit/pull request is solely related to CairoLS, please prefix your commit message/PR title
-with `LS: `.
+The latest stable Rust is the only thing you really need.
+It is recommended to use [rustup](https://rustup.rs/) for getting it.
+
+This document assumes that you are using Visual Studio Code with
+the [Cairo extension](https://marketplace.visualstudio.com/items?itemName=starkware.cairo1) as the
+editor of choice for running CairoLS against.
+You might also want to check out
```
[the remark this answers is not in the document]
Comment (mkaput) [reply CONTRIBUTING.md:26]: I'm mentioning `.settings/vscode.json` further down this file

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -1,27 +1,38 @@
...
-We are building an extensive end-to-end tests suite for CairoLS [here](./tests/e2e).
+We are building an extensive end-to-end tests suite for
+CairoLS [here](crates/cairo-lang-language-server/tests/e2e).
 These tests implement a simple language client that you can control (like put a cursor at certain
 position, send a request to the server, etc.).
 Check out existing tests for examples of what you can do.
-If you need, do not hesitate to extend the language client with new capabilities!
-Its source code is located [here](./tests/e2e/support/mod.rs).
+If you need, don’t hesitate to extend the language client with new capabilities!
+Its source code is located [here](crates/cairo-lang-language-server/tests/e2e/support/mod.rs).
```
[the remark this answers is not in the document]
Comment (mkaput) [reply CONTRIBUTING.md:37]: Ugh, markdown files are not link-checked (and never were). Good catch, fixed this link.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,27 +1,38 @@
...
+
+When contributing to this repository, please first discuss the change you wish to make via issue,
+email, or any other method with the owners of this repository before making a change.
 
-## Commits
+## Environment setup
 
-If your commit/pull request is solely related to CairoLS, please prefix your commit message/PR title
-with `LS: `.
+The latest stable Rust is the only thing you really need.
```
[the remark this answers is not in the document]
Comment (mkaput) [reply CONTRIBUTING.md]: 50/50, I deliberately tried to minimize them. Furthermore, I'm mentioning in next commits how you HAVE to have Scarb installed in order to work and I need some piece of doc to link to

File: .github/workflows/ci.yml (hunk 2 above)
Comment (mkaput) [reply .github/workflows/ci.yml:65]: we're using nightly rustfmt settings, they live in separate section in rustfmt.toml

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mkaput) [reply .github/workflows/ci.yml:19]: I want to run test on Windows in the future

File: src/project/unmanaged_core_crate.rs (hunk 3 above)
Comment (mkaput) [reply src/project/unmanaged_core_crate.rs:41]: nooo. this repository does not have corelib anymore

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Arcticae) [reply .github/workflows/ci.yml:19]: premature optimization but i can live with that

File: CONTRIBUTING.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (Arcticae) [reply CONTRIBUTING.md:26]: Ah ok

File: Cargo.toml (hunk 6 above)
[the remark this answers is not in the document]
Comment (Arcticae) [reply Cargo.toml:83]: perhaps it's for the better right now

**Review by Arcticae — APPROVED** [review]

**Pull request merged**